### PR TITLE
contribute to specs.md

### PR DESCRIPTION
**Pull request by AdedamolaXL**

### CODE REVIEW

File: specs.md [hunk 1]
```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
...
+The main changes in the genesis iteration are chainId, timestamp and the withdrawal credentials of the first validator.
+    
+The network always starts from a genesis which is deterministically created based on the original one, genesis_0. Expiration of the genesis is given by its `genesis_timestamp` and `period` (a predefined constant, length of time a single ephemeral network runs). Therfore once the currenttimestamp reaches the terminal time of ephemeral network, it has to switch to a new genesis. This can be expressed as:
 
-Given a known `period` (the length of time network instance will run), the client can calculate the number of lifecycle iterations from `genesis_0` to the current one and create a new genesis with latest parameters. Always, when the client starts with an option of ephemeral testnet, it checks the current genesis timestamp and identifies whether it's older than `timestamp + period`, at which point it triggers generation of a new genesis.
+    let terminal_timestamp = genesis_timestamp + period
```
[the remark this answers is not in the document]
Comment (AdedamolaXL) [reply specs.md:28]: no i am not, `terminal_timestamp` is me rewriting the previous paragraph in pseudocode so someone reading the specifications for the first time can understand better

which honestly was the whole point of the pr update in the first place :)

**Review by flexibrah — APPROVED** [review]

File: specs.md [hunk 2]
```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
...
+    
+    if (current_timestamp >= terminal_timestamp):
+       return new_genesis
+    else:
+       return none
```
Comment (taxmeifyoucan) [inline specs.md:33]: This part should be part of  the reset function described below. Terminal timestamp is not considered in genesis function, only in the reset

File: specs.md [hunk 3]
```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
...
 
-New genesis needs to be created and correspond in both EL and CL client. Manual implementation of this mechanism can be found in [this script](https://hackmd.io/@jimmygchen/rJw2Zjl2j). 
+Clients shall include a hardcoded genesis_0, similarly to other predefined networks. But this genesis is only used at the very beginning of the testnet existence, in its first iteration 0. Later on, with iteration 1 and further, client does not initialize this genesis but uses it to derive the current one. 
+
+Thus, for any iteration i, after the first iteration (i > 0):
+
+    genesis_timestamp_i = period * i + genesis_timestamp_0.
```
Comment (taxmeifyoucan) [inline specs.md:39]: First step is to calculate `i` based on timestamps

File: specs.md [hunk 4]
```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
...
+
+When the client starts with the option of ephemeral testnet, it checks whether a genesis for the network is present. If it doesn't exist or the current genesis timestamp is older than genesis_timestamp + period, it triggers the generation of a new genesis. This new genesis, derived from the genesis 0, will be written to the database and used to run the current network.
+
+      let genesis = genesis_from_database()
+      
+      if (genesis = 0 || current_timestamp >= min_genesis_time + period):
```
Comment (taxmeifyoucan) [inline specs.md:45]: not sure what pseudocode convention are you using, maybe it's correct, just that `genesis = 0` as non existent seems a bit confusing here. I would change it to NaN, null or !

File: specs.md [hunk 5]
```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
...
+         generate new_genesis 
+         return genesis_from_database(new_genesis)
+      else:
+         return genesis_from_database()
+
+The choice of period can be very important and have certain implications on netowrk stability and overall user experience.
```
Comment (taxmeifyoucan) [inline specs.md:51]: The period reasoning is explained below, you can remove this part or add it below

File: specs.md [hunk 6]
```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
...
+   * Clear documentation that helps users understand the need for frequenct changes and how to effectively manage them.
+   * A user notification system, like a simple countdown showing how much time before the upcoming network reset.
+   
 
 #### Consensus client
 
-Genesis generation in CL client uses the same parameters as EL but requires also updated genesis state in ssz. 
+Genesis generation in CL client uses the same parameters as EL, while also requiring the generation of the updated genesis state in SSZ format, with minor changes which include:
+
+* A deposit (smart) contract responsible for managing the process of validators making deposits to participate in the network.
```
Comment (taxmeifyoucan) [inline specs.md:106]: The deposit contract is on EL and doesn't change, CL just needs to know its address, deployment hight etc which stay constant

File: specs.md [hunk 7]
```diff
@@ -19,50 +19,140 @@ This approach is based on data gathered from public automatically resetted netwo
...
+  
+   * During each iteration, the CL client generates a new set of validators based on trusted entities within the community.
```
Comment (taxmeifyoucan) [inline specs.md:108]: The validator set stays the same, otherwise clients would need to create a release to update the set

**Review by taxmeifyoucan — COMMENTED** [review]
thanks for your updates! I left some comments but overall, I don't recommend paying much attention to these specs as it's going to be deprecated in favor of EIP.

**Pull request closed without merging**